pkg/receive/handler.go

```diff
@@ -408,9 +410,15 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 	span, ctx := tracing.StartSpan(r.Context(), "receive_http")
 	defer span.Finish()
 
-	tenant := r.Header.Get(h.options.TenantHeader)
+	tenant := r.Header.Get(path.Base(h.options.TenantHeader))
+	// Avoids malicious override of the tenant when using path.Base
+	if tenant != r.Header.Get(h.options.TenantHeader) {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
 	if tenant == "" {
-		tenant = h.options.DefaultTenantID
+		tenant = path.Base(h.options.DefaultTenantID)
 	}
 
```

Comment:
I think that the logic can be simplified so that there is less room for error if we keep the original code as-is and then add the following four lines right here. We should determine the tenant value like we did originally and _then_ sanitize it. Otherwise, we are doing `path.Base` too many times, which spreads out our sanitation logic and makes it more likely that we will miss one of the lines in future maintenance:
```golang
if tenant != path.Base(tenant) {
	http.Error(w, err.Error(), http.StatusInternalServerError)
	return
}
```

Reply:
Yep, totally agree, still new to the codebase so thanks for pointing that out. I'll update the PR.

Reply:
Thanks, that's better! Thinking about it more, we should actually move this sanitization down below the next `if` block where we get the tenant from the certificate.

Maybe we even want to extract tenant value extraction and sanitization into its own, isolated function to keep the logic very tight.

Reply:
Thanks! I think @GiedriusS recommended something similar too. 🙂 

I wonder if we should use `path.Base` at all here, though. In case the tenant header value is set to something like `/`, we can still have `path.Base("/") == "/"`. Maybe instead just checking if the string contains any `/` would be easier to check here?